content/src/service/ServiceImpl.ts

```diff
@@ -341,51 +350,14 @@ export class ServiceImpl implements MetaverseContentService {
   }
 
   async getEntitiesByIds(ids: EntityId[]): Promise<Entity[]> {
-    const deployments = await getDeployments(this.components, {
-      filters: { entityIds: ids, onlyCurrentlyPointed: true }
-    })
-    return this.mapDeploymentsToEntities(deployments)
+    return this.components.activeEntities.withIds(...ids)
   }
 
-  async getEntitiesByPointers(type: EntityType, pointers: Pointer[]): Promise<Entity[]> {
-    const allEntities = await this.cache.get(type, pointers, async (type, pointers) => {
-      const deployments = await getDeployments(this.components, {
-        filters: { entityTypes: [type], pointers, onlyCurrentlyPointed: true }
-      })
-
-      const entities = this.mapDeploymentsToEntities(deployments)
-      const entries: [Pointer, Entity][][] = entities.map((entity) =>
-        entity.pointers.map((pointer) => [pointer, entity])
-      )
-
-      const pointersMap = new Map<Pointer, Entity | undefined>(entries.flat())
-
-      // Get Deployments only retrieves the active entities, so if a pointer has a null value we need to manually define it
-      for (const pointer of pointers) {
-        if (!pointersMap.has(pointer)) pointersMap.set(pointer, undefined)
-      }
-      return pointersMap
-    })
-
-    // Since the same entity might appear many times, we must remove duplicates
-    const grouped = new Map(allEntities.map((entity) => [entity.id, entity]))
-    return Array.from(grouped.values())
-  }
-
-  private mapDeploymentsToEntities(history: PartialDeploymentHistory<Deployment>): Entity[] {
-    return history.deployments.map(
-      ({ entityVersion, entityId, entityType, pointers, entityTimestamp, content, metadata }) => ({
-        version: entityVersion,
-        id: entityId,
-        type: entityType,
-        pointers,
-        timestamp: entityTimestamp,
-        content: content?.map(({ key, hash }) => ({ file: key, hash })),
-        metadata
-      })
-    )
+  async getEntitiesByPointers(pointers: Pointer[]): Promise<Entity[]> {
+    return this.components.activeEntities.withPointers(...pointers)
```

Comment:
You don't need to destructure the ids since the method already receives an array